examples/va/nfv/nfv-ovs-dpdk-sriov-hci/edpm-pre-ceph/nodeset/values.yaml

```diff
@@ -0,0 +1,281 @@
+                mtu: 9000
+                rx_queue: 2
+                members:
+                  - type: interface
+                    name: nic5
+          - type: sriov_pf
+            name: nic6
+            numvfs: 10
+            mtu: 9000
+            use_dhcp: false
+            promisc: true
+          - type: sriov_pf
+            name: nic7
+            numvfs: 10
+            mtu: 9000
+            use_dhcp: false
+            promisc: true
+        # These vars are for the network config templates themselves and are
+        # considered EDPM network defaults.
+        # edpm_nodes_validation
+        edpm_nodes_validation_validate_controllers_icmp: false
+        edpm_nodes_validation_validate_gateway_icmp: false
+        dns_search_domains: []
+        gather_facts: false
+        # edpm firewall, change the allowed CIDR if needed
+        edpm_sshd_configure_firewall: true
+        edpm_sshd_allowed_ranges:
+          - 192.168.122.0/24
+        edpm_ceph_hci_pre_enabled_services:
+          - ceph_mon
+          - ceph_mgr
+          - ceph_osd
+          - ceph_rgw
+          - ceph_nfs
+          - ceph_rgw_frontend
+          - ceph_nfs_frontend
+        storage_mtu: 9000
+        storage_mgmt_mtu: 9000
+        storage_mgmt_vlan_id: 145
+        storage_mgmt_cidr: "24"
+        storage_mgmt_host_routes: []
+    networks:
+      - defaultRoute: true
+        name: ctlplane
+        subnetName: subnet1
+      - name: internalapi
+        subnetName: subnet1
+      - name: storage
+        subnetName: subnet1
+      - name: storagemgmt
+        subnetName: subnet1
+      - name: tenant
+        subnetName: subnet1
+    nodes:
+      edpm-compute-0:
+        ansible:
+          ansibleHost: 192.168.122.100
+        hostName: edpm-compute-0
+        networks:
+          - defaultRoute: true
+            fixedIP: 192.168.122.100
+            name: ctlplane
+            subnetName: subnet1
+          - name: internalapi
+            subnetName: subnet1
+          - name: storage
+            subnetName: subnet1
+          - name: storagemgmt
+            subnetName: subnet1
+          - name: tenant
+            subnetName: subnet1
+      edpm-compute-1:
+        ansible:
+          ansibleHost: 192.168.122.101
+        hostName: edpm-compute-1
+        networks:
+          - defaultRoute: true
+            fixedIP: 192.168.122.101
+            name: ctlplane
+            subnetName: subnet1
+          - name: internalapi
+            subnetName: subnet1
+          - name: storage
+            subnetName: subnet1
+          - name: storagemgmt
+            subnetName: subnet1
+          - name: tenant
+            subnetName: subnet1
+    services:
+      - bootstrap
+      - download-cache
+      - reboot-os
+      - configure-ovs-dpdk
+      - configure-network
+      - validate-network
+      - install-os
+      - ceph-hci-pre
+      - configure-os
+      - ssh-known-hosts
+      - run-os
+      - install-certs
```

Comment:
Few services mentioned in post ceph are duplicated in post ceph also. can we remove here.

Reply:
Deployment is done in the following way:
- preceph: ovs-dpdk-sriov working deployment  (as in ovs-dpdk-sriov architecture)
- postceph: Add hci to a ovs-dpdk-sriov deployment (as in hci architecture)
I would keep duplicated services for the following reason:
- This architecture is a mix of ovs-dpdk-sriov and hci architectures, so for maintenance reasons I would keep services as in those architectures.
- Preceph generates an working ovs-dpdk-sriov. That deployment is updated with ceph and some services are executed again in order to update it

Reply:
@fultonj Will it work to first fully deploy the dataplane services with the pre-Ceph stage and then later install Ceph and redeploy the services again to use Ceph?  I assume we have a good reason for leaving out some of the dataplane services in pre-Ceph stage of the HCI VA, which are instead included here.

Reply:
@abays I already tested it and it worked. I first deployed a nfv cluster and them modified it with hci. I did in this way because of 2 reasons:
1. allows to test step by step and isolate issues. If there are issues with nfv deployment, hci will probably fail as well.
2. allows to keep nfv and hci changes differenciated. nfv changes are in preceph and hci in postceph. So, nfv-hci templates will be easier to update if either nfv or hci templates are updated.

Reply:
I understand it was tested and that it worked  in that the deployment finished and passed testing, but it's computationally wasteful which will make the job take longer than it has to for each run. It is also not a good example to have in our architecture repo. Why run ansible tasks which are known to NOT bring the system to the desired state only to run them later?

Here are the lists for HCI:

1. pre-ceph https://github.com/openstack-k8s-operators/architecture/blob/main/examples/va/hci/edpm-pre-ceph/nodeset/values.yaml#L157-L166
2. post-ceph https://github.com/openstack-k8s-operators/architecture/blob/main/examples/va/hci/values.yaml#L11-L17

Similarly your pre-list is right here (line 29 of `examples/va/nfv/nfv-ovs-dpdk-sriov-hci/edpm-pre-ceph/nodeset/values.yaml`) and your post list is here:

https://github.com/openstack-k8s-operators/architecture/pull/316/files#diff-8acddd6371172de43cf1bd274f9c694e8f877a826ab8eeee4d4a5e7d5abc19c4R11

Would you please  try with these two lists:

```yaml
services:  # pre-ceph
  - bootstrap
  - download-cache
  - reboot-os
  - configure-ovs-dpdk
  - configure-network
  - validate-network
  - install-os
  - ceph-hci-pre
  - configure-os
  - ssh-known-hosts
  - run-os

services:  # post-ceph
  - install-certs
  - ceph-client
  - ovn
  - neutron-ovn-igmp
  - neutron-metadata
  - neutron-metadata
  - neutron-sriov
  - libvirt
  - nova-custom-ovsdpdksriov          # can this be merged into the "-ceph" service?
  - nova-custom-ovsdpdksriov-ceph
  - telemetry
```

Reply:
ok, i will test it